charts/kube-prometheus-stack/templates/alertmanager/networkpolicy.yaml

```diff
@@ -0,0 +1,80 @@
+{{- if and .Values.alertmanager.enabled .Values.alertmanager.networkPolicy.enabled }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ template "kube-prometheus-stack.fullname" . }}-alertmanager
+  namespace: {{ template "kube-prometheus-stack.namespace" . }}
+  labels:
+    app: {{ template "kube-prometheus-stack.name" . }}-alertmanager
+    {{- include "kube-prometheus-stack.labels" . | nindent 4 }}
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: alertmanager
+  policyTypes:
+    {{- toYaml .Values.alertmanager.networkPolicy.policyTypes | nindent 4 }}
+  ingress:
+    # Allow ingress from gateway
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: {{ .Values.alertmanager.networkPolicy.gateway.namespace }}
+          podSelector:
+            matchLabels:
+              {{- toYaml .Values.alertmanager.networkPolicy.gateway.podLabels | nindent 14 }}
```

Comment:
Question: what happens, if `.Values.alertmanager.networkPolicy.gateway.podLabels` is empty? Do we need to cover this scenario and omit the podSelector, if podLabels are empty?

Reply:
Modified the template to omit podSelector when podLabels is empty or null. Added a test case that verifies this behavior by setting podLabels: null. The template now correctly handles cases where:

- podLabels is null
- podLabels is empty ({})
- podLabels is not set